woodwork/statistics_utils.py

```diff
@@ -343,6 +343,136 @@ def _get_value_counts(dataframe, ascending=False, top_n=10, dropna=False):
     return val_counts
 
 
+def _calculate_iqr_bounds(series=None, quantiles=None):
+    """Calculates bounds for outlier detection using the 1.5*IQR method.
+
+    Args:
+        series (pd.Series, optional): Data used to calculate first and third quartiles used to calcualte
+            the interquartile range. If present, quartiles will be calculated even if passed in.
+        quantiles (dict[float->float], optional): Quantiles that can be used to calculate the interquartile
+            range. The keys of the dictionary should be the quantile floating point value.
```

Comment:
Should this also be quartiles? If we don't pass in quartiles, won't we get key errors when we try to index with `quantiles[0.25]` and `quantiles[0.75]`? Unless of course, the quantiles happen to contain 0.25 and 0.75 values.

Sorry to nit-pick a docstring on a private function, but maybe the docstring should indicate that if quantiles are passed without a series, the values need to include 0.25 and 0.75 values.

Reply:
You bring up a good point about when we should be using "quantiles" vs "quartiles". We want users to be able to pass through any quantiles they'd like (0.0 and 1.0 being the main ones since, in theory, those could be used to build a slightly different box plot than one that uses the high and low bounds as whiskers).

However, we need at least the quartiles 0.25 and 0.75 to perform the calculations in these functions. I think that the docstrings here probably need to be a lot more clear about what the minimum requirements are, and what values will be returned.